api/src/main/java/javax/enterprise/inject/spi/InjectionTargetFactory.java

```diff
@@ -56,4 +59,19 @@
      */
     public InjectionTarget<T> createInjectionTarget(Bean<T> bean);
 
+    /**
+     *
+     * Returns an {@link AnnotatedTypeConfigurator} to add or remove injection point to the {@link InjectionTarget} to create.
+     * {@link #createInjectionTarget(Bean)} will use the {@link AnnotatedTypeConfigurator} to define injection points in the returned
+     * {@link InjectionTarget}
+     *
+     * Each call returns the same AnnotatedTypeConfigurator.
+     *
+     * @return an {@link AnnotatedTypeConfigurator} to configure injection points
+     * @since 2.0
+     */
+    default AnnotatedTypeConfigurator<T> configure() {
+        return null;
```

Comment:
I don't think this is good. In fact, returning `null` in default impl seems rather nasty.
I can understand that you meant to provide a default method, however this class is not implemented by end user but rather by the CDI impl itself, therefore *not having* it `default` wouldn't cause so much harm IMO.